lib/model/folder.go

```diff
@@ -161,7 +171,7 @@ func (f *folder) Serve(ctx context.Context) error {
 
 	// If we're configured to not do version cleanup, or we don't have a
 	// versioner, cancel and drain that timer now.
-	if f.versionCleanupInterval == 0 || f.versioner == nil {
+	if f.versionCleanupInterval == 0 || ((f.versionType == "trashcan" || f.versionType == "simple") && f.versionCleanoutDays == 0) || (f.versionType == "staggered" && f.versionMaxAge == 0) || f.versionType == "external" || f.versioner == nil {
```

Comment:
This stuff could in fact happen in the constructor instead, then we don't need to have instance variables for all this stuff and it would keep things more contained, I think... We could also break up and comment the very long if clause a little bit.